# Asterisk R2 patch for brazilian networks
# robert@marestelecom.net
#
# info clipped from http://www.soft-switch.org/unicall/installing-mfcr2.html :
# ---8<---
# A number of installations in Brazil require a timeout for clearing a call. 
# This is to prevent them from clearing when the telco sends metering pulses. 
# To enable this, look for the following line:
#         mfcr2->clear_back_persistence_check = 0;
#
# and change it to:
#         mfcr2->clear_back_persistence_check = 300;
#
# and recompile the software. This timeout is in milliseconds. 
# If you still have trouble, trying increasing it to 500. 
# Future versions of Unicall will make this timeout selectable without recompiling.
# --->8---

--- old/mfcr2.c	2005-11-11 22:10:09.000000000 -0300
+++ new/mfcr2.c	2007-03-28 11:22:21.000000000 -0300
@@ -2973,7 +2973,7 @@ static void load_r2_parameter_set(mfcr2_
     mfcr2->inbound_cas_persistence_check = 20;
     mfcr2->outbound_cas_persistence_check = 20;
     mfcr2->max_seize_ack_wait = DEFAULT_MAX_SEIZE_ACK_WAIT;
-    mfcr2->clear_back_persistence_check = 0;
+    mfcr2->clear_back_persistence_check = 300;
 
     /* For several variants of MFC/R2, the forward I-15 tone signals the
        end of ANI digits, but not the end of DNIS digits. If the inbound
